keras_nlp/src/models/resnet/resnet_backbone.py

```diff
@@ -139,21 +144,64 @@ def __init__(
         else:
             x = image_input
 
-        x = layers.Conv2D(
-            64,
-            7,
-            strides=2,
-            padding="same",
-            data_format=data_format,
-            use_bias=use_pre_activation,
-            dtype=dtype,
-            name="conv1_conv",
-        )(x)
-        if not use_pre_activation:
+        if version == "vd":
+            # ResNet_vd
+            x = layers.Conv2D(
+                32,
+                3,
+                strides=2,
+                padding="same",
+                use_bias=False,
+                dtype=dtype,
+                name="conv1_conv",
+            )(x)
             x = layers.BatchNormalization(
                 axis=bn_axis, epsilon=1.001e-5, dtype=dtype, name="conv1_bn"
             )(x)
             x = layers.Activation("relu", dtype=dtype, name="conv1_relu")(x)
+            x = layers.Conv2D(
```

Comment:
A colab verifying the numerics for this would be helpful the verify architecture correctness

Reply:
Modifications compared to plain ResNetV1 are
* Average pooling is used in shortcut connections instead of CNN strides
* The model's first convolutional layer is replaced with three subsequent ones
* Stride configuration in residual blocks is slightly different

[This paper](https://www.researchgate.net/publication/349646156_Automatic_Detection_and_Classification_of_Steel_Surface_Defect_Using_Deep_Convolutional_Neural_Networks) has some good visualizations of the ResNet_vd architecture.

Reply:
I can look into creating a notebook to verify numerics. Since we are primarily adding this for the differential binarization model, I would check numerics based on the pretrained differential binarization model from PaddleOCR unless you think we should compare numerics for the classifier network as well.

Reply:
you can verify the numerics just for the resnetvd backbone with just random weights.